fakestorage/object.go

```diff
@@ -74,8 +74,8 @@ func (s *Server) createObject(obj Object) error {
 
 // ListObjects returns a sorted list of objects that match the given criteria,
 // or an error if the bucket doesn't exist.
-func (s *Server) ListObjects(bucketName, prefix, delimiter string) ([]Object, []string, error) {
-	backendObjects, err := s.backend.ListObjects(bucketName)
+func (s *Server) ListObjects(bucketName, prefix, delimiter string, versions bool) ([]Object, []string, error) {
```

Comment:
Should we introduce a ``ListObjectsWithOpts`` and deprecate this method?

Reply:
Hi!! Francisco, completely up to you (and I can change if you think is convenient), but I think there is an important difference between both functions: `CreateBucket` / `CreateBucketWithOpts` are probably used directly by users of this project: they are actually not published / mapped in the router of the server (but `createBucketByPost`, that calls directly createBucket in the backend - something maybe we want to change), and I guess most of the people (including me) are using that to generate scenarios/prepare a bucket status to run tests against (as the code you want to test normally expects the bucket to be already available as part of the infra/deployment setup). However, I guess `ListObjects` is mostly consumed via the API we expose in the server, so through a gcs client, and its rare people invokes it directly to do assertions... that means changes in this contract are probably not as severe as the CreateBucket, that I bet is called directly from test code. 

Do you have the same view than me? Best!!!!

Reply:
That's fair enough. Let's merge as is. Thanks for the quick response! 😁